tests/baselines/reference/genericCallWithNonSymmetricSubtypes.errors.txt

```diff
@@ -1,48 +0,0 @@
-tests/cases/conformance/types/typeRelationships/typeInference/genericCallWithNonSymmetricSubtypes.ts(12,9): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
-  Type argument candidate '{ x: number; y?: number; }' is not a valid type argument because it is not a supertype of candidate '{ x: number; z?: number; }'.
-    Property 'y' is missing in type '{ x: number; z?: number; }'.
-tests/cases/conformance/types/typeRelationships/typeInference/genericCallWithNonSymmetricSubtypes.ts(13,10): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
-  Type argument candidate '{ x: number; z?: number; }' is not a valid type argument because it is not a supertype of candidate '{ x: number; y?: number; }'.
-    Property 'z' is missing in type '{ x: number; y?: number; }'.
-
-
-==== tests/cases/conformance/types/typeRelationships/typeInference/genericCallWithNonSymmetricSubtypes.ts (2 errors) ====
-    // generic type argument inference where inference leads to two candidates that are both supertypes of all candidates
-    // we choose the first candidate so the result is dependent on the order of the arguments provided
-    
-    function foo<T>(x: T, y: T) {
-        var r: T;
-        return r;
-    }
-    
-    var a: { x: number; y?: number; };
-    var b: { x: number; z?: number; };
-    
-    var r = foo(a, b); // { x: number; y?: number; };

```

Comment:
This doesn't seem right - there's no error, and the type of `r` is pretty arbitrary now. It seems like it'd be better to infer from the intersection of the remaining supertypes - this would give `r` the correct type (`{ x: number; y?: number} & { x: number; z?: number }`), and the leaf-node error messages will still be good.